osgp/shared/osgp-ws-smartmetering/src/main/resources/schemas/sm-configuration.xsd

```diff
@@ -1238,13 +1238,60 @@
     <xsd:complexContent>
       <xsd:extension base="common:Command">
         <xsd:sequence>
-          <xsd:element name="directAttach" type="xsd:int"/>
-          <xsd:element name="randomisationStartWindow" type="xsd:int"/>
-          <xsd:element name="multiplicationFactor" type="xsd:int"/>
-          <xsd:element name="numberOfRetries" type="xsd:int"/>
+          <xsd:element name="directAttach" type="xsd:int" />
+          <xsd:element name="randomisationStartWindow" type="xsd:int" />
+          <xsd:element name="multiplicationFactor" type="xsd:int" />
+          <xsd:element name="numberOfRetries" type="xsd:int" />
         </xsd:sequence>
       </xsd:extension>
     </xsd:complexContent>
   </xsd:complexType>
 
+  <xsd:element name="GetUnencryptedKeysRequest">
+    <xsd:complexType>
+      <xsd:sequence>
+        <xsd:element type="tns:SecretType" name="secretType"
+          minOccurs="1" maxOccurs="4" />
+      </xsd:sequence>
+    </xsd:complexType>
+  </xsd:element>
+
+  <xsd:element name="GetUnencryptedKeysAsyncRequest">
+    <xsd:complexType>
+      <xsd:complexContent>
+        <xsd:extension base="common:AsyncRequest" />
+      </xsd:complexContent>
+    </xsd:complexType>
+  </xsd:element>
+
+  <xsd:element name="GetUnencryptedKeysResponse">
+    <xsd:complexType>
+      <xsd:sequence>
+        <xsd:element type="tns:SecretType" name="secretType" />
+        <xsd:element type="xsd:string" name="secretValue" />
```

Comment:
It would probably be good to treat key data the same across operations.
In `sm-installation.xsd` some of the keys appear in complex type `Device` where they are included as `xsd:hexBinary`.
Another location where keys appear as `hexBinary` is in this same XSD, with complex type `SetKeysRequestData`.
The choice for `hexBinary` has the added benefit of being more explicit on how to interpret the data to get to the bytes.

Reply:
ok, thanks for pointing that out.